Source/DafnyCore/AST/Expressions/Expressions.cs

```diff
@@ -700,6 +703,47 @@ public abstract class Expression : TokenNode {
     return ite;
   }
 
+  /// <summary>
+  /// Create a resolved function-call expression. The returned expression will have syntactic scaffolding, which
+  /// enables resolving a syntactic clone of this resolved expression.
+  /// Expects "receiver" to be a resolved expression.
+  /// Expects that "function" has no type parameters, but it's okay for the enclosing type to have type parameters.
```

Comment:
What if it does have a type parameter?

Reply:
Then, either this method will need to be expanded to handle that case, too, or we'd need a different method for it. For now, the only use of this `CreateResolvedCall` method is by `CreateResolvedValidCall` (which is part of the `:autocontract` machinery), and `Valid()` does not have any type parameters.